From c53e835509deb18e93c8ae6d750beaa70312276c Mon Sep 17 00:00:00 2001
From: Patrick Mooney <pmooney@pfmooney.com>
Date: Mon, 31 Oct 2016 22:10:44 +0000
Subject: [PATCH] OS-5756 lx_tgkill holds p_lock on return, killing machine
 Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com> Reviewed by: Bryan
 Cantrill <bryan@joyent.com> Approved by: Bryan Cantrill <bryan@joyent.com>

---
 usr/src/uts/common/brand/lx/syscall/lx_kill.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/usr/src/uts/common/brand/lx/syscall/lx_kill.c b/usr/src/uts/common/brand/lx/syscall/lx_kill.c
index 851ecdc610..032ef44dbc 100644
--- a/usr/src/uts/common/brand/lx/syscall/lx_kill.c
+++ b/usr/src/uts/common/brand/lx/syscall/lx_kill.c
@@ -117,6 +117,7 @@ lx_thrkill(pid_t tgid, pid_t pid, int lx_sig, boolean_t tgkill)
 	if (tgkill) {
 		if ((pid == 1 && tgid != 1) ||
 		    (pid != 1 && tgid != pp->p_pid)) {
+			mutex_exit(&pp->p_lock);
 			rv = set_errno(ESRCH);
 			goto free_and_exit;
 		}
-- 
2.21.0

